commands/command_clone.go

```diff
@@ -79,8 +81,36 @@ func cloneCommand(cmd *cobra.Command, args []string) {
 		fetchRef("HEAD", include, exclude)
 	} else {
 		pull(include, exclude)
+
+		err := postCloneSubmodules(args)
+		if err != nil {
+			Exit("Error performing 'git lfs pull' for submodules: %v", err)
+		}
+	}
+
+}
+
+func postCloneSubmodules(args []string) error {
+	// In git 2.9+ the filter option will have been passed through to submodules
+	// So we need to lfs pull inside each, if the original clone command
+	if !git.Config.IsGitVersionAtLeast("2.9.0") {
+		// In earlier versions submodules would have used smudge filter
+		return nil
+	}
+	// Also we only do this if --recursive or --recurse-submodules was provided
+	if !cloneFlags.Recursive && !cloneFlags.RecurseSubmodules {
+		return nil
 	}
 
+	// Use `git submodule foreach --recursive` to cascade into nested submodules
+	// Also good to call a new instance of git-lfs rather than do things
+	// inside this instance, since that way we get a clean env in that subrepo
+	cmd := subprocess.ExecCommand("git", "submodule", "foreach", "--recursive",
```

Comment:
As a thought, we could support the `--jobs` flag from the [`git-clone`](https://git-scm.com/docs/git-clone) documentation.

Instead of running `git submodule foreach`, we could just get a list of _all_ submodule paths, feed all of those paths into a channel, and then slurp them from N goroutines. As a first pass, `--jobs` could be set to 1 so that it effectively behaves the same. I'm thinking something like:

```go
var wg sync.WaitGroup
var paths chan string

modules, err := subprocess.ExecCommand("git", "get-all-submodule-paths")
if err != nil {
        // ...
}

wg.Add(len(modules))
go func() {
        for _, path := range modules {
                paths <- path
        }
}()

var jobs int
for i range jobs {
        go func(paths <-chan string) {
        select {
        case path := <-paths:
                // run git-lfs pull
                wg.Done()
        }
        }
}

wg.Wait()
```